pkg/controllers/node/termination/terminator/eviction.go

```diff
@@ -69,122 +66,97 @@ func IsNodeDrainError(err error) bool {
 	return errors.As(err, &nodeDrainErr)
 }
 
-type QueueKey struct {
-	types.NamespacedName
-	UID        types.UID
-	providerID string
-}
-
-func NewQueueKey(pod *corev1.Pod, providerID string) QueueKey {
-	return QueueKey{
-		NamespacedName: client.ObjectKeyFromObject(pod),
-		UID:            pod.UID,
-		providerID:     providerID,
-	}
-}
-
 type Queue struct {
-	workqueue.TypedRateLimitingInterface[QueueKey]
+	sync.Mutex
 
-	mu  sync.Mutex
-	set sets.Set[QueueKey]
+	source chan event.TypedGenericEvent[*corev1.Pod]
```

Comment:
Should we make the channel buffered to avoid having to wait on the reconciler pulling items off of the queue before we proceed?